app/controllers/post_types_controller.rb

```diff
@@ -10,4 +11,36 @@ def create
 
     redirect_back fallback_location: group_url, alert: post_type.errors.to_a.first
   end
+
+  def index
+    @post_types = @group.post_types.select do |post_type|
+      ![
+        Membership::ECONOMIST_ID,
+        Membership::LEADER_POST_ID,
+        Membership::PAST_LEADER_ID,
+        Membership::DEFAULT_POST_ID,
+        Membership::PEK_ADMIN_ID,
+        Membership::NEW_MEMBER_ID
+      ].include? post_type.id
+    end
+  end
+
+  def destroy
+    @post_type = PostType.find(params[:id])
+    active = @group.memberships.any? do |membership|
+      membership.posts.any? do |post|
+        post.post_type.id == @post_type.id
+      end
+    end
```

Comment:
I feel this is a bit too much here. I think it's enough to check if the post type has any related post, isn't? To achieve this you can set the opposite of [this relation](https://github.com/kir-dev/pek-next/blob/master/app/models/post.rb#L8) in `PostType` model, like `has_many :posts, foreign_key: :pttip_id` and you can use the `post_type.posts.any?` to verify if it could be deleted.

Also I feel some lack of security constraints here. We should check if the post type is related with a group and the user is the group's leader. Now I think all the common types are not connected with any group, but I'd add a check if the group id refers a common type or not. This last part I let you decide.

Reply:
For the first half, you're right, I'll refactor.

For the second: I think the `before_action :require_leader` takes care of the authorization process (only allow group leaders, the tests also indicated this behavior).
Well the common types are mean to be non-destructible (they don't appear on the editing UI), so I guess adding an extra security layer is always welcomed.

Reply:
As you mention, you are almost right with the `require_leader`.. but I can send a request with a group that I'm leader of and with a totally different group's post type. Nothing validates their relation.

You can check if the post type also related with the current group OR I forgot to mention, but instead of nested routes it is advised to use shallow ones and with that you can check if the user is the leader of the given post type. I'll let you decide this too, for the second one it would be much more ideal if we'd have pundit.